# 🚛 Análisis de Siniestros Nacionales
### Sector de Recolección de Residuos

---

**Descripción del Proyecto:**

Este notebook contiene el análisis exploratorio de datos (EDA) completo del dataset de siniestros nacionales del sector de recolección de residuos en Colombia. El objetivo es identificar patrones, tendencias y factores de riesgo para mejorar la seguridad vial y la eficiencia operativa.

**Contenido:**
1. Importación de librerías y carga de datos
2. Limpieza y transformación de datos
3. Análisis exploratorio de datos (EDA)
4. Análisis temporal
5. Análisis geográfico
6. Análisis de causas
7. Análisis de vehículos
8. Análisis de conductores
9. Visualizaciones avanzadas
10. Conclusiones y recomendaciones

**Autor:** Equipo de Análisis de Datos

**Fecha:** Marzo 2025

---

## 📚 1. Importación de Librerías

In [1]:
# Librerías para manipulación de datos
import numpy as np
import pandas as pd
from datetime import datetime, timedelta

# Librerías para visualización
import matplotlib.pyplot as plt
import seaborn as sns
import plotly.graph_objects as go
import plotly.express as px
from plotly.subplots import make_subplots

# Configuración de visualización
plt.style.use('seaborn-v0_8-darkgrid')
sns.set_palette("husl")
%matplotlib inline

# Configuración de pandas
pd.set_option('display.max_columns', None)
pd.set_option('display.max_rows', 100)
pd.set_option('display.float_format', '{:.2f}'.format)

# Suprimir warnings
import warnings
warnings.filterwarnings('ignore')

print("✅ Librerías importadas correctamente")

✅ Librerías importadas correctamente


## 📂 2. Carga de Datos

In [2]:
# Cargar el dataset de siniestros
df = pd.read_csv("Siniestros_nacionales_aleatorio_trim6.csv", encoding='latin-1', sep=';')

print(f"✅ Dataset cargado correctamente")
print(f"📊 Dimensiones: {df.shape[0]} filas x {df.shape[1]} columnas")

✅ Dataset cargado correctamente
📊 Dimensiones: 1133 filas x 26 columnas


In [3]:
# Vista previa de los datos
print("\n📋 PRIMERAS 5 FILAS:")
df.head()


📋 PRIMERAS 5 FILAS:


,DEPARTAMENTO,CIUDAD,Responsabilidad de la empresa,Área operativa,Personal capacitado,Fecha Siniestro,Tipo de vehículo,Placa Vehículo,Cédula Conductor,Nombre Conductor,Culpabilidad Conductor,Causa Siniestro,Origen siniestro,Hipótesis,Tipo accidente,Acciones al conductor,Servicio,Turno,Reportado Aseguradora,Asistencia Jurídica,Usó Grua,Asistencia Tránsito,Tiempo Inoperativo ( HORAS ),Descripción de Causa,Descripción Daños,Información Adicional
0,ATLANTICO,SOLEDAD,NO,Operaciones,SI,31/08/2022,Volqueta,SXF858,93397030-PINEDA GARRIDO RAFAEL HUMBERTO,JUANJOSE MAESTRE PAEZ,NO,FACTORES EXTERNOS,Conductor,No mantener distancia de seguridad,Colisiones,Descargos,Recolección Barrido Manual y Empuje,Diurno,SI,SI,NO,NO,2,"SIENDO LAS 18:30 DE LA NOCHE, EN LA CRA 8 CON ...",DAÑO RETROVISOR VOLQUETA PROPIA,SE REALIZÓ LLAMADO DE INMEDIATO A LA LÍNEA DE ...
1,ATLANTICO,SANTO TOMAS,NO,Operaciones,SI,14/12/2024,Compactador,LKN387,98603456-RESTREPO VASQUEZ WILSON ARMANDO,ARBOLEDA GARZON JEAN PIERRE,NO,FACTORES EXTERNOS,Conductor,Otra,Daños infraestructura de,Descargos,Recolección Domiciliaria,Diurno,NO,NO,NO,NO,1,"A LAS 15:20, LA COMPACTADORA LKN852, CON CP225...",DAÑO EN EL PORTON DE LA UNIDAD,SE REALIZA COMUNICACIÓN CON LA ADMINISTRACION ...
2,ATLANTICO,PONEDERA,SI,Operaciones,NO,24/11/2021,Compactador,TNH223,98575351-GIRALDO GUTIERREZ ALVARO,ANYENSON EMERIO MANZANEDA CARRILLO,NO,FALLAS HUMANAS,Vehículo,Otra,Daños infraestructura de,Descargos,Recolección Domiciliaria,Diurno,SI,SI,NO,NO,1,SIENDO APROXIMADAMENTE LAS 09:00 AM LA COMPACT...,"POSTE DE ENERGIA, CON BASE DETERIORADA",SE REALIZO EL REPORTE A LA EMPRESA DE ENERGÍA ...
3,ATLANTICO,JUAN DE ACOSTA,SI,Operaciones,NO,07/12/2021,Compactador,FNL700,8466098-RIVERA ALVAREZ MILTON CESAR,RENDÓN JHON ALEXANDER,SI,FALLAS HUMANAS,Conductor,No mantener distancia de seguridad,Colisiones,Descargos,Recolección Domiciliaria,Diurno,SI,SI,NO,SI,2,SIENDO LAS 12:10 PM EL CONDUCTOR ASIGNADO AL ...,GOLPE EN LA PARTE TRASERA DEL BUS,LA SUPERVISORA LORENA ARIAS OJEDA QUE SE ENCON...
4,ATLANTICO,SOLEDAD,SI,Operaciones,NO,14/12/2021,Compactador,SDT814,98520623-IDARRAGA VELEZ JOSE ROMAN,FLOREZ AVENDAÑO ARAMANDO ALEXANDER,SI,FALLAS HUMANAS,Conductor,No mantener distancia de seguridad,Colisiones,Descargos,Recolección Domiciliaria,Diurno,SI,SI,NO,SI,2,SE NOTIFICA A LA EMPRESA ASEGURADORA #888,DAÑO DEFENSA TRASERA PARTE IZQUIERDA,SIENDO APROXIMADAMENTE LAS 9:38 AM DEL 14 DE D...


In [4]:
# Información general del dataset
print("\n📊 INFORMACIÓN GENERAL:")
df.info()


📊 INFORMACIÓN GENERAL:
<class 'pandas.DataFrame'>
RangeIndex: 1133 entries, 0 to 1132
Data columns (total 26 columns):
 #   Column                         Non-Null Count  Dtype
---  ------                         --------------  -----
 0   DEPARTAMENTO                   1133 non-null   str  
 1   CIUDAD                         1133 non-null   str  
 2   Responsabilidad de la empresa  1133 non-null   str  
 3   Área operativa                 1133 non-null   str  
 4   Personal capacitado            1133 non-null   str  
 5   Fecha Siniestro                1133 non-null   str  
 6   Tipo de vehículo               1132 non-null   str  
 7   Placa Vehículo                 1133 non-null   str  
 8   Cédula Conductor               1133 non-null   str  
 9   Nombre Conductor               1133 non-null   str  
 10  Culpabilidad Conductor         1133 non-null   str  
 11  Causa Siniestro                1133 non-null   str  
 12  Origen siniestro               1133 non-null   str  
 13  Hipót

## 🧹 3. Limpieza y Transformación de Datos

In [6]:
# Crear una copia del dataframe original
df_original = df.copy()

# Limpiar nombres de columnas (eliminar espacios extras)
df.columns = df.columns.str.strip()

print("✅ Nombres de columnas limpiados")

✅ Nombres de columnas limpiados


In [7]:
# Verificar y convertir tipos de datos
print("\n🔧 CONVERSIÓN DE TIPOS DE DATOS:")

# Convertir fecha a datetime si no lo está
if df['Fecha Siniestro'].dtype != 'datetime64[ns]':
    df['Fecha Siniestro'] = pd.to_datetime(df['Fecha Siniestro'])
    print("✅ Columna 'Fecha Siniestro' convertida a datetime")

# Verificar duplicados
duplicados = df.duplicated().sum()
print(f"\n🔍 Registros duplicados encontrados: {duplicados}")

if duplicados > 0:
    df.drop_duplicates(inplace=True)
    print(f"✅ {duplicados} duplicados eliminados")


🔧 CONVERSIÓN DE TIPOS DE DATOS:
✅ Columna 'Fecha Siniestro' convertida a datetime

🔍 Registros duplicados encontrados: 0


In [10]:
# Análisis de valores nulos
print("\n❌ VALORES NULOS POR COLUMNA:")
nulos = df.isnull().sum()
nulos_pct = (nulos / len(df)) * 100

nulos_df = pd.DataFrame({
    'Columna': nulos.index,
    'Valores Nulos': nulos.values,
    'Porcentaje': nulos_pct.values
}).sort_values('Valores Nulos', ascending=False)

nulos_df[nulos_df['Valores Nulos'] > 0]


❌ VALORES NULOS POR COLUMNA:


,Columna,Valores Nulos,Porcentaje
25,Información Adicional,690,60.90
24,Descripción Daños,5,0.44
6,Tipo de vehículo,1,0.09
23,Descripción de Causa,1,0.09


In [11]:
# Crear columnas derivadas para análisis temporal
print("\n🕐 CREANDO COLUMNAS TEMPORALES:")

df['Año'] = df['Fecha Siniestro'].dt.year
df['Mes'] = df['Fecha Siniestro'].dt.month
df['Mes_Nombre'] = df['Fecha Siniestro'].dt.month_name()
df['Día'] = df['Fecha Siniestro'].dt.day
df['Día_Semana'] = df['Fecha Siniestro'].dt.day_name()
df['Día_Semana_Num'] = df['Fecha Siniestro'].dt.dayofweek
df['Trimestre'] = df['Fecha Siniestro'].dt.quarter
df['Semana_Año'] = df['Fecha Siniestro'].dt.isocalendar().week
df['Semestre'] = df['Mes'].apply(lambda x: 'S1' if x <= 6 else 'S2')
df['Año_Mes'] = df['Fecha Siniestro'].dt.to_period('M')

print("✅ Columnas temporales creadas:")
print("   - Año, Mes, Mes_Nombre, Día")
print("   - Día_Semana, Día_Semana_Num")
print("   - Trimestre, Semestre, Semana_Año")
print("   - Año_Mes")


🕐 CREANDO COLUMNAS TEMPORALES:
✅ Columnas temporales creadas:
   - Año, Mes, Mes_Nombre, Día
   - Día_Semana, Día_Semana_Num
   - Trimestre, Semestre, Semana_Año
   - Año_Mes


In [13]:
# Crear categorías de gravedad basadas en tiempo inoperativo
print("\n⚠️ CREANDO CATEGORÍA DE GRAVEDAD:")

df['Gravedad'] = pd.cut(
    df['Tiempo Inoperativo ( HORAS )'],
    bins=[-np.inf, 0, 2, 24, 100, np.inf],
    labels=['Sin tiempo', 'Leve', 'Moderado', 'Grave', 'Crítico']
)

print("✅ Categoría de gravedad creada:")
print("   - Sin tiempo: 0 horas")
print("   - Leve: 1-2 horas")
print("   - Moderado: 3-24 horas")
print("   - Grave: 25-100 horas")
print("   - Crítico: >100 horas")

print("\n📊 Distribución de Gravedad:")
print(df['Gravedad'].value_counts())


⚠️ CREANDO CATEGORÍA DE GRAVEDAD:
✅ Categoría de gravedad creada:
   - Sin tiempo: 0 horas
   - Leve: 1-2 horas
   - Moderado: 3-24 horas
   - Grave: 25-100 horas
   - Crítico: >100 horas

📊 Distribución de Gravedad:
Gravedad
Leve          817
Sin tiempo    165
Moderado      138
Grave           7
Crítico         6
Name: count, dtype: int64


In [15]:
# Crear indicador de fin de semana
df['Es_FinDeSemana'] = df['Día_Semana_Num'].apply(lambda x: 'Sí' if x >= 5 else 'No')

# Crear indicador de horario laboral (aproximado)
# Asumimos que el turno nocturno es de mayor riesgo
df['Turno_Riesgo'] = df['Turno'].apply(lambda x: 'Alto' if x == 'Nocturno' else 'Normal')

print("\n✅ Variables adicionales creadas:")
print("   - Es_FinDeSemana")
print("   - Turno_Riesgo")


✅ Variables adicionales creadas:
   - Es_FinDeSemana
   - Turno_Riesgo


In [17]:
# Resumen final del dataset limpio
print("\n" + "="*80)
print("📊 RESUMEN FINAL DEL DATASET LIMPIO")
print("="*80)
print(f"\n📏 Dimensiones: {df.shape[0]} filas x {df.shape[1]} columnas")
print(f"📅 Periodo: {df['Fecha Siniestro'].min().strftime('%d/%m/%Y')} - {df['Fecha Siniestro'].max().strftime('%d/%m/%Y')}")
print(f"🗺️ Departamentos únicos: {df['DEPARTAMENTO'].nunique()}")
print(f"🏙️ Ciudades únicas: {df['CIUDAD'].nunique()}")
print(f"🚗 Tipos de vehículos: {df['Tipo de vehículo'].nunique()}")
print(f"👤 Conductores únicos: {df['Nombre Conductor'].nunique()}")
print(f"\n✅ Datos listos para análisis")


📊 RESUMEN FINAL DEL DATASET LIMPIO

📏 Dimensiones: 1133 filas x 39 columnas
📅 Periodo: 02/06/2021 - 13/11/2025
🗺️ Departamentos únicos: 4
🏙️ Ciudades únicas: 45
🚗 Tipos de vehículos: 15
👤 Conductores únicos: 787

✅ Datos listos para análisis


## 📊 4. Análisis Exploratorio de Datos (EDA)

### 4.1 Estadísticas Descriptivas Generales

In [19]:
# Estadísticas de tiempo inoperativo
print("⏱️ ESTADÍSTICAS DE TIEMPO INOPERATIVO (HORAS):")
print("="*60)

tiempo_stats = df['Tiempo Inoperativo ( HORAS )'].describe()
print(tiempo_stats)

print(f"\n📊 Estadísticas adicionales:")
print(f"   Suma total: {df['Tiempo Inoperativo ( HORAS )'].sum():,.0f} horas")
print(f"   Rango: {df['Tiempo Inoperativo ( HORAS )'].max() - df['Tiempo Inoperativo ( HORAS )'].min():,.0f} horas")
print(f"   Percentil 90: {df['Tiempo Inoperativo ( HORAS )'].quantile(0.90):.2f} horas")
print(f"   Percentil 95: {df['Tiempo Inoperativo ( HORAS )'].quantile(0.95):.2f} horas")

⏱️ ESTADÍSTICAS DE TIEMPO INOPERATIVO (HORAS):
count   1133.00
mean       8.36
std      132.94
min       -3.00
25%        1.00
50%        1.00
75%        2.00
max     3600.00
Name: Tiempo Inoperativo ( HORAS ), dtype: float64

📊 Estadísticas adicionales:
   Suma total: 9,473 horas
   Rango: 3,603 horas
   Percentil 90: 3.00 horas
   Percentil 95: 5.00 horas


In [21]:
# KPIs principales
print("\n🎯 INDICADORES CLAVE DE DESEMPEÑO (KPIs):")
print("="*60)

total_siniestros = len(df)
conductor_culpable = (df['Culpabilidad Conductor'] == 'SI').sum()
pct_culpable = (conductor_culpable / total_siniestros) * 100
reportados = (df['Reportado Aseguradora'] == 'SI').sum()
pct_reportados = (reportados / total_siniestros) * 100
uso_grua = (df['Usó Grua'] == 'SI').sum()
pct_grua = (uso_grua / total_siniestros) * 100
asistencia_juridica = (df['Asistencia Jurídica'] == 'SI').sum()
pct_juridica = (asistencia_juridica / total_siniestros) * 100

print(f"📌 Total de Siniestros: {total_siniestros:,}")
print(f"👤 Conductor Culpable: {conductor_culpable:,} ({pct_culpable:.1f}%)")
print(f"📋 Reportados a Aseguradora: {reportados:,} ({pct_reportados:.1f}%)")
print(f"🚧 Uso de Grúa: {uso_grua:,} ({pct_grua:.1f}%)")
print(f"⚖️ Asistencia Jurídica: {asistencia_juridica:,} ({pct_juridica:.1f}%)")
print(f"⏱️ Tiempo Inoperativo Promedio: {df['Tiempo Inoperativo ( HORAS )'].mean():.2f} horas")


🎯 INDICADORES CLAVE DE DESEMPEÑO (KPIs):
📌 Total de Siniestros: 1,133
👤 Conductor Culpable: 509 (44.9%)
📋 Reportados a Aseguradora: 814 (71.8%)
🚧 Uso de Grúa: 82 (7.2%)
⚖️ Asistencia Jurídica: 627 (55.3%)
⏱️ Tiempo Inoperativo Promedio: 8.36 horas


### 4.2 Distribución de Variables Categóricas

In [22]:
# Distribución por departamento
print("\n🗺️ DISTRIBUCIÓN POR DEPARTAMENTO:")
print("="*60)

dept_counts = df['DEPARTAMENTO'].value_counts()
dept_pct = (dept_counts / len(df)) * 100

dept_df = pd.DataFrame({
    'Cantidad': dept_counts,
    'Porcentaje': dept_pct
})
print(dept_df)


🗺️ DISTRIBUCIÓN POR DEPARTAMENTO:
              Cantidad  Porcentaje
DEPARTAMENTO                      
ANTIOQUIA          536       47.31
TOLIMA             277       24.45
CESAR              202       17.83
ATLANTICO          118       10.41


In [23]:
# Top 15 ciudades con más siniestros
print("\n🏙️ TOP 15 CIUDADES CON MÁS SINIESTROS:")
print("="*60)

ciudad_counts = df['CIUDAD'].value_counts().head(15)
ciudad_pct = (ciudad_counts / len(df)) * 100

ciudad_df = pd.DataFrame({
    'Cantidad': ciudad_counts,
    'Porcentaje': ciudad_pct
})
print(ciudad_df)


🏙️ TOP 15 CIUDADES CON MÁS SINIESTROS:
             Cantidad  Porcentaje
CIUDAD                           
IBAGUE            260       22.95
BELLO             210       18.53
VALLEDUPAR        147       12.97
ITAGUI            132       11.65
SOLEDAD            69        6.09
SABANETA           45        3.97
COPACABANA         37        3.27
CALDAS             25        2.21
CODAZZI            25        2.21
LA ESTRELLA        21        1.85
GIRARDOTA          18        1.59
MALAMBO            11        0.97
FREDONIA           10        0.88
AMBALEMA           10        0.88
SAN PEDRO           9        0.79


In [24]:
# Distribución por tipo de vehículo
print("\n🚗 DISTRIBUCIÓN POR TIPO DE VEHÍCULO:")
print("="*60)

vehiculo_counts = df['Tipo de vehículo'].value_counts()
vehiculo_pct = (vehiculo_counts / len(df)) * 100

vehiculo_df = pd.DataFrame({
    'Cantidad': vehiculo_counts,
    'Porcentaje': vehiculo_pct
})
print(vehiculo_df)


🚗 DISTRIBUCIÓN POR TIPO DE VEHÍCULO:
                  Cantidad  Porcentaje
Tipo de vehículo                      
Compactador            805       71.05
Volqueta               133       11.74
Amplirolles             57        5.03
Camion                  44        3.88
Buseta                  36        3.18
Cabezotes               20        1.77
Furgon < 3 Ton          12        1.06
Barredora                7        0.62
Multi container          5        0.44
Camioneta                4        0.35
Motocarro                3        0.26
Carro Tanque             2        0.18
Cargador                 2        0.18
13                       1        0.09
Retroexcavadora          1        0.09


In [25]:
# Distribución por tipo de accidente
print("\n💥 DISTRIBUCIÓN POR TIPO DE ACCIDENTE:")
print("="*60)

accidente_counts = df['Tipo accidente'].value_counts().head(10)
accidente_pct = (accidente_counts / len(df)) * 100

accidente_df = pd.DataFrame({
    'Cantidad': accidente_counts,
    'Porcentaje': accidente_pct
})
print(accidente_df)


💥 DISTRIBUCIÓN POR TIPO DE ACCIDENTE:
                              Cantidad  Porcentaje
Tipo accidente                                    
Colisiones                         444       39.19
Daños a vehículo tercero (No       226       19.95
Daños infraestructura de           143       12.62
Redes electricas                   111        9.80
Daños a vehículo propio (No        104        9.18
Lesiones personales terceros        30        2.65
terceros (No incluye redes          28        2.47
Daños infraestructura propia        11        0.97
Pérdida total del vehículo          10        0.88
incluye colisión)                   10        0.88


In [26]:
# Distribución por causa de siniestro
print("\n🔍 DISTRIBUCIÓN POR CAUSA DE SINIESTRO:")
print("="*60)

causa_counts = df['Causa Siniestro'].value_counts()
causa_pct = (causa_counts / len(df)) * 100

causa_df = pd.DataFrame({
    'Cantidad': causa_counts,
    'Porcentaje': causa_pct
})
print(causa_df)

# Gráfico de pastel para causas
fig = go.Figure(data=[go.Pie(
    labels=causa_counts.index,
    values=causa_counts.values,
    hole=0.4,
    marker=dict(colors=['#FF6B6B', '#4ECDC4', '#45B7D1'])
)])

fig.update_layout(
    title='Distribución de Causas de Siniestros',
    title_font_size=20,
    height=500
)
fig.show()


🔍 DISTRIBUCIÓN POR CAUSA DE SINIESTRO:
                   Cantidad  Porcentaje
Causa Siniestro                        
FACTORES EXTERNOS       617       54.46
FALLAS HUMANAS          466       41.13
FALLAS MTTO              50        4.41


### 4.3 Visualización de Distribuciones

In [27]:
# Histograma de tiempo inoperativo
fig = px.histogram(
    df[df['Tiempo Inoperativo ( HORAS )'] < 100],  # Filtrar outliers extremos
    x='Tiempo Inoperativo ( HORAS )',
    nbins=50,
    title='Distribución de Tiempo Inoperativo (< 100 horas)',
    labels={'Tiempo Inoperativo ( HORAS )': 'Horas Inoperativas'},
    color_discrete_sequence=['#3498db']
)

fig.update_layout(
    xaxis_title='Tiempo Inoperativo (horas)',
    yaxis_title='Frecuencia',
    showlegend=False,
    height=500
)
fig.show()

In [28]:
# Gráfico de barras: Siniestros por departamento
dept_counts = df['DEPARTAMENTO'].value_counts()

fig = go.Figure(data=[go.Bar(
    x=dept_counts.index,
    y=dept_counts.values,
    marker_color=['#e74c3c', '#3498db', '#2ecc71', '#f39c12'],
    text=dept_counts.values,
    textposition='outside'
)])

fig.update_layout(
    title='Siniestros por Departamento',
    xaxis_title='Departamento',
    yaxis_title='Número de Siniestros',
    template='plotly_white',
    height=500
)
fig.show()

In [29]:
# Gráfico de barras horizontales: Top 10 ciudades
top_ciudades = df['CIUDAD'].value_counts().head(10)

fig = go.Figure(data=[go.Bar(
    x=top_ciudades.values,
    y=top_ciudades.index,
    orientation='h',
    marker_color='lightblue',
    text=top_ciudades.values,
    textposition='outside'
)])

fig.update_layout(
    title='Top 10 Ciudades con Más Siniestros',
    xaxis_title='Número de Siniestros',
    yaxis_title='Ciudad',
    template='plotly_white',
    height=500
)
fig.show()

## 📅 5. Análisis Temporal

### 5.1 Evolución Anual

In [30]:
# Siniestros por año
print("\n📊 SINIESTROS POR AÑO:")
print("="*60)

año_counts = df['Año'].value_counts().sort_index()
print(año_counts)

# Gráfico de línea temporal
fig = go.Figure()

fig.add_trace(go.Scatter(
    x=año_counts.index,
    y=año_counts.values,
    mode='lines+markers',
    name='Siniestros',
    line=dict(color='#3498db', width=3),
    marker=dict(size=10)
))

fig.update_layout(
    title='Evolución Anual de Siniestros',
    xaxis_title='Año',
    yaxis_title='Número de Siniestros',
    template='plotly_white',
    height=500,
    hovermode='x unified'
)
fig.show()


📊 SINIESTROS POR AÑO:
Año
2021     66
2022    329
2023    296
2024    226
2025    216
Name: count, dtype: int64


In [31]:
# Estadísticas por año
print("\n📊 RESUMEN ESTADÍSTICO POR AÑO:")
print("="*60)

resumen_anual = df.groupby('Año').agg({
    'CIUDAD': 'count',
    'Tiempo Inoperativo ( HORAS )': ['mean', 'median', 'sum'],
    'Culpabilidad Conductor': lambda x: (x == 'SI').sum(),
    'Reportado Aseguradora': lambda x: (x == 'SI').sum()
}).round(2)

resumen_anual.columns = ['Total', 'Tiempo_Prom', 'Tiempo_Mediana', 'Tiempo_Total', 'Conductor_Culpable', 'Reportados']
print(resumen_anual)


📊 RESUMEN ESTADÍSTICO POR AÑO:
      Total  Tiempo_Prom  Tiempo_Mediana  Tiempo_Total  Conductor_Culpable  \
Año                                                                          
2021     66         2.29            1.00           151                  33   
2022    329         9.89            1.00          3255                 160   
2023    296         2.97            1.00           879                 138   
2024    226        19.65            1.00          4442                  88   
2025    216         3.45            1.00           746                  90   

      Reportados  
Año               
2021          50  
2022         232  
2023         212  
2024         177  
2025         143  


### 5.2 Patrones Mensuales

In [32]:
# Siniestros por mes (todos los años combinados)
mes_counts = df.groupby('Mes').size().sort_index()
meses_nombres = ['Ene', 'Feb', 'Mar', 'Abr', 'May', 'Jun', 
                 'Jul', 'Ago', 'Sep', 'Oct', 'Nov', 'Dic']

fig = go.Figure()

fig.add_trace(go.Bar(
    x=meses_nombres,
    y=mes_counts.values,
    marker_color='lightcoral',
    text=mes_counts.values,
    textposition='outside'
))

fig.update_layout(
    title='Distribución de Siniestros por Mes (Todos los Años)',
    xaxis_title='Mes',
    yaxis_title='Total de Siniestros',
    template='plotly_white',
    height=500
)
fig.show()

In [33]:
# Heatmap: Siniestros por Año y Mes
heatmap_data = df.groupby(['Año', 'Mes']).size().unstack(fill_value=0)

fig = go.Figure(data=go.Heatmap(
    z=heatmap_data.values,
    x=meses_nombres,
    y=heatmap_data.index,
    colorscale='Blues',
    text=heatmap_data.values,
    texttemplate='%{text}',
    textfont={"size": 12}
))

fig.update_layout(
    title='Mapa de Calor: Siniestros por Año y Mes',
    xaxis_title='Mes',
    yaxis_title='Año',
    height=500
)
fig.show()

### 5.3 Análisis por Día de la Semana

In [34]:
# Siniestros por día de la semana
orden_dias_en = ['Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday', 'Sunday']
dias_es = ['Lunes', 'Martes', 'Miércoles', 'Jueves', 'Viernes', 'Sábado', 'Domingo']

dia_counts = df['Día_Semana'].value_counts().reindex(orden_dias_en, fill_value=0)

fig = go.Figure()

fig.add_trace(go.Bar(
    x=dias_es,
    y=dia_counts.values,
    marker_color=['#3498db', '#3498db', '#3498db', '#3498db', '#3498db', '#e74c3c', '#e74c3c'],
    text=dia_counts.values,
    textposition='outside'
))

fig.update_layout(
    title='Siniestros por Día de la Semana',
    xaxis_title='Día',
    yaxis_title='Número de Siniestros',
    template='plotly_white',
    height=500
)
fig.show()

print("\n📊 Estadísticas por día:")
for dia_en, dia_es in zip(orden_dias_en, dias_es):
    count = dia_counts[dia_en]
    pct = (count / len(df)) * 100
    print(f"   {dia_es}: {count} ({pct:.1f}%)")


📊 Estadísticas por día:
   Lunes: 163 (14.4%)
   Martes: 186 (16.4%)
   Miércoles: 184 (16.2%)
   Jueves: 163 (14.4%)
   Viernes: 182 (16.1%)
   Sábado: 198 (17.5%)
   Domingo: 57 (5.0%)


### 5.4 Análisis por Turno

In [35]:
# Comparativa entre turnos
turno_stats = df.groupby('Turno').agg({
    'CIUDAD': 'count',
    'Tiempo Inoperativo ( HORAS )': 'mean',
    'Culpabilidad Conductor': lambda x: (x == 'SI').sum() / len(x) * 100,
    'Usó Grua': lambda x: (x == 'SI').sum() / len(x) * 100
}).round(2)

turno_stats.columns = ['Total', 'Tiempo_Promedio', 'Pct_Culpabilidad', 'Pct_Grua']

print("\n🌓 COMPARATIVA POR TURNO:")
print("="*60)
print(turno_stats)

# Gráfico comparativo
fig = make_subplots(
    rows=1, cols=2,
    subplot_titles=('Distribución por Turno', '% Culpabilidad del Conductor'),
    specs=[[{'type': 'pie'}, {'type': 'bar'}]]
)

turno_counts = df['Turno'].value_counts()

fig.add_trace(
    go.Pie(labels=turno_counts.index, values=turno_counts.values, hole=0.4),
    row=1, col=1
)

fig.add_trace(
    go.Bar(x=turno_stats.index, y=turno_stats['Pct_Culpabilidad'], 
           marker_color=['#3498db', '#e74c3c']),
    row=1, col=2
)

fig.update_layout(height=500, showlegend=True)
fig.show()


🌓 COMPARATIVA POR TURNO:
          Total  Tiempo_Promedio  Pct_Culpabilidad  Pct_Grua
Turno                                                       
Diurno      688            12.22             42.01      7.12
Nocturno    445             2.39             49.44      7.42


## 🗺️ 6. Análisis Geográfico

### 6.1 Análisis por Departamento

In [36]:
# Resumen detallado por departamento
print("\n🗺️ RESUMEN POR DEPARTAMENTO:")
print("="*80)

dept_detail = df.groupby('DEPARTAMENTO').agg({
    'CIUDAD': ['count', 'nunique'],
    'Tiempo Inoperativo ( HORAS )': ['mean', 'median', 'max'],
    'Culpabilidad Conductor': lambda x: (x == 'SI').sum(),
    'Reportado Aseguradora': lambda x: (x == 'SI').sum(),
    'Usó Grua': lambda x: (x == 'SI').sum()
}).round(2)

dept_detail.columns = ['Total_Siniestros', 'Ciudades_Afectadas', 'Tiempo_Prom', 
                       'Tiempo_Mediana', 'Tiempo_Max', 'Conductor_Culpable', 
                       'Reportados', 'Uso_Grua']
print(dept_detail)


🗺️ RESUMEN POR DEPARTAMENTO:
              Total_Siniestros  Ciudades_Afectadas  Tiempo_Prom  \
DEPARTAMENTO                                                      
ANTIOQUIA                  536                  15         5.99   
ATLANTICO                  118                  15         6.83   
CESAR                      202                  10        24.29   
TOLIMA                     277                   5         2.00   

              Tiempo_Mediana  Tiempo_Max  Conductor_Culpable  Reportados  \
DEPARTAMENTO                                                               
ANTIOQUIA               1.00        2592                 303         449   
ATLANTICO               2.00         384                  46          80   
CESAR                   1.00        3600                  70         147   
TOLIMA                  2.00         100                  90         138   

              Uso_Grua  
DEPARTAMENTO            
ANTIOQUIA           28  
ATLANTICO           11  
CESAR     

In [37]:
# Gráfico de barras apiladas: Causas por Departamento
causa_dept = pd.crosstab(df['DEPARTAMENTO'], df['Causa Siniestro'])

fig = go.Figure()

for causa in causa_dept.columns:
    fig.add_trace(go.Bar(
        name=causa,
        x=causa_dept.index,
        y=causa_dept[causa]
    ))

fig.update_layout(
    title='Causas de Siniestros por Departamento',
    xaxis_title='Departamento',
    yaxis_title='Número de Siniestros',
    barmode='stack',
    template='plotly_white',
    height=500
)
fig.show()

### 6.2 Análisis por Ciudad

In [38]:
# Top 20 ciudades con análisis detallado
print("\n🏙️ TOP 20 CIUDADES - ANÁLISIS DETALLADO:")
print("="*80)

top_20_ciudades = df['CIUDAD'].value_counts().head(20).index

ciudad_detail = df[df['CIUDAD'].isin(top_20_ciudades)].groupby('CIUDAD').agg({
    'DEPARTAMENTO': 'first',
    'Fecha Siniestro': 'count',
    'Tiempo Inoperativo ( HORAS )': ['mean', 'max'],
    'Culpabilidad Conductor': lambda x: (x == 'SI').sum(),
    'Tipo de vehículo': lambda x: x.mode()[0] if not x.mode().empty else 'N/A'
}).round(2)

ciudad_detail.columns = ['Departamento', 'Total', 'Tiempo_Prom', 'Tiempo_Max', 
                         'Conductor_Culpable', 'Vehiculo_Comun']
ciudad_detail = ciudad_detail.sort_values('Total', ascending=False)
print(ciudad_detail)


🏙️ TOP 20 CIUDADES - ANÁLISIS DETALLADO:
                   Departamento  Total  Tiempo_Prom  Tiempo_Max  \
CIUDAD                                                            
IBAGUE                   TOLIMA    260         2.05         100   
BELLO                 ANTIOQUIA    210        13.50        2592   
VALLEDUPAR                CESAR    147        30.29        3600   
ITAGUI                ANTIOQUIA    132         0.92           3   
SOLEDAD               ATLANTICO     69         2.28          10   
SABANETA              ANTIOQUIA     45         1.20           6   
COPACABANA            ANTIOQUIA     37         1.16           5   
CODAZZI                   CESAR     25         2.00          13   
CALDAS                ANTIOQUIA     25         2.04          18   
LA ESTRELLA           ANTIOQUIA     21         1.05           2   
GIRARDOTA             ANTIOQUIA     18         1.28           7   
MALAMBO               ATLANTICO     11        11.27         100   
AMBALEMA            

In [39]:
# Mapa de árbol (Treemap): Ciudades por Departamento
ciudad_dept_counts = df.groupby(['DEPARTAMENTO', 'CIUDAD']).size().reset_index(name='Cantidad')
top_ciudades_treemap = ciudad_dept_counts.nlargest(30, 'Cantidad')

fig = px.treemap(
    top_ciudades_treemap,
    path=['DEPARTAMENTO', 'CIUDAD'],
    values='Cantidad',
    title='Top 30 Ciudades Agrupadas por Departamento',
    color='Cantidad',
    color_continuous_scale='Blues'
)

fig.update_layout(height=600)
fig.show()

## 🔍 7. Análisis de Causas

In [40]:
# Análisis detallado por causa
print("\n🔍 ANÁLISIS DETALLADO POR CAUSA:")
print("="*80)

for causa in df['Causa Siniestro'].unique():
    df_causa = df[df['Causa Siniestro'] == causa]
    
    print(f"\n📌 {causa}")
    print("-" * 60)
    print(f"   Total de casos: {len(df_causa):,}")
    print(f"   Porcentaje del total: {len(df_causa)/len(df)*100:.1f}%")
    print(f"   Tiempo inoperativo promedio: {df_causa['Tiempo Inoperativo ( HORAS )'].mean():.2f} horas")
    print(f"   Tiempo inoperativo mediana: {df_causa['Tiempo Inoperativo ( HORAS )'].median():.2f} horas")
    print(f"   Conductor culpable: {(df_causa['Culpabilidad Conductor'] == 'SI').sum()} ({(df_causa['Culpabilidad Conductor'] == 'SI').sum()/len(df_causa)*100:.1f}%)")
    print(f"   Departamento más afectado: {df_causa['DEPARTAMENTO'].mode()[0]}")
    print(f"   Tipo de accidente más común: {df_causa['Tipo accidente'].mode()[0]}")


🔍 ANÁLISIS DETALLADO POR CAUSA:

📌 FACTORES EXTERNOS
------------------------------------------------------------
   Total de casos: 617
   Porcentaje del total: 54.5%
   Tiempo inoperativo promedio: 12.28 horas
   Tiempo inoperativo mediana: 1.00 horas
   Conductor culpable: 101 (16.4%)
   Departamento más afectado: ANTIOQUIA
   Tipo de accidente más común: Colisiones

📌 FALLAS HUMANAS
------------------------------------------------------------
   Total de casos: 466
   Porcentaje del total: 41.1%
   Tiempo inoperativo promedio: 1.69 horas
   Tiempo inoperativo mediana: 1.00 horas
   Conductor culpable: 402 (86.3%)
   Departamento más afectado: ANTIOQUIA
   Tipo de accidente más común: Colisiones

📌 FALLAS MTTO
------------------------------------------------------------
   Total de casos: 50
   Porcentaje del total: 4.4%
   Tiempo inoperativo promedio: 22.12 horas
   Tiempo inoperativo mediana: 2.00 horas
   Conductor culpable: 6 (12.0%)
   Departamento más afectado: ANTIOQUIA
   T

In [41]:
# Comparativa de causas en el tiempo
causas_tiempo = df.groupby(['Año', 'Causa Siniestro']).size().reset_index(name='Cantidad')

fig = px.line(
    causas_tiempo,
    x='Año',
    y='Cantidad',
    color='Causa Siniestro',
    title='Evolución de Causas de Siniestros por Año',
    markers=True,
    template='plotly_white'
)

fig.update_layout(
    xaxis_title='Año',
    yaxis_title='Número de Siniestros',
    height=500,
    hovermode='x unified'
)
fig.show()

## 🚗 8. Análisis de Vehículos

In [42]:
# Análisis por tipo de vehículo
print("\n🚗 ANÁLISIS POR TIPO DE VEHÍCULO:")
print("="*80)

vehiculo_stats = df.groupby('Tipo de vehículo').agg({
    'CIUDAD': 'count',
    'Tiempo Inoperativo ( HORAS )': ['mean', 'median'],
    'Culpabilidad Conductor': lambda x: (x == 'SI').sum() / len(x) * 100,
    'Usó Grua': lambda x: (x == 'SI').sum() / len(x) * 100,
    'Reportado Aseguradora': lambda x: (x == 'SI').sum() / len(x) * 100
}).round(2)

vehiculo_stats.columns = ['Total', 'Tiempo_Prom', 'Tiempo_Mediana', 
                          'Pct_Culpabilidad', 'Pct_Grua', 'Pct_Reportado']
vehiculo_stats = vehiculo_stats.sort_values('Total', ascending=False)
print(vehiculo_stats)


🚗 ANÁLISIS POR TIPO DE VEHÍCULO:
                  Total  Tiempo_Prom  Tiempo_Mediana  Pct_Culpabilidad  \
Tipo de vehículo                                                         
Compactador         805         2.40            1.00             45.59   
Volqueta            133         8.21            1.00             45.86   
Amplirolles          57         1.60            1.00             38.60   
Camion               44        59.93            1.00             47.73   
Buseta               36         1.25            1.00             36.11   
Cabezotes            20         1.20            1.00             25.00   
Furgon < 3 Ton       12         0.75            0.50             66.67   
Barredora             7         3.00            2.00             28.57   
Multi container       5         1.60            1.00             80.00   
Camioneta             4       900.75            1.00             25.00   
Motocarro             3         1.33            1.00             66.67   
Carg

In [43]:
# Gráfico de comparación: Top 8 vehículos
top_8_vehiculos = vehiculo_stats.head(8)

fig = make_subplots(
    rows=2, cols=2,
    subplot_titles=('Total Siniestros', 'Tiempo Promedio Inoperativo', 
                    '% Culpabilidad', '% Uso de Grúa'),
    specs=[[{'type': 'bar'}, {'type': 'bar'}],
           [{'type': 'bar'}, {'type': 'bar'}]]
)

fig.add_trace(
    go.Bar(x=top_8_vehiculos.index, y=top_8_vehiculos['Total'], name='Total', marker_color='lightblue'),
    row=1, col=1
)

fig.add_trace(
    go.Bar(x=top_8_vehiculos.index, y=top_8_vehiculos['Tiempo_Prom'], name='Tiempo Prom', marker_color='lightcoral'),
    row=1, col=2
)

fig.add_trace(
    go.Bar(x=top_8_vehiculos.index, y=top_8_vehiculos['Pct_Culpabilidad'], name='% Culpabilidad', marker_color='lightgreen'),
    row=2, col=1
)

fig.add_trace(
    go.Bar(x=top_8_vehiculos.index, y=top_8_vehiculos['Pct_Grua'], name='% Grúa', marker_color='lightyellow'),
    row=2, col=2
)

fig.update_layout(height=800, showlegend=False, title_text="Análisis Comparativo por Tipo de Vehículo (Top 8)")
fig.show()

## 👤 9. Análisis de Conductores

In [44]:
# Top 20 conductores con más siniestros
print("\n👤 TOP 20 CONDUCTORES CON MÁS SINIESTROS:")
print("="*80)

conductor_counts = df['Nombre Conductor'].value_counts().head(20)

conductor_detail = df[df['Nombre Conductor'].isin(conductor_counts.index)].groupby('Nombre Conductor').agg({
    'Fecha Siniestro': 'count',
    'Culpabilidad Conductor': lambda x: (x == 'SI').sum(),
    'Tiempo Inoperativo ( HORAS )': 'sum',
    'Tipo de vehículo': lambda x: x.mode()[0] if not x.mode().empty else 'Varios'
})

conductor_detail.columns = ['Total_Siniestros', 'Veces_Culpable', 'Tiempo_Total_Inop', 'Vehiculo_Usual']
conductor_detail = conductor_detail.sort_values('Total_Siniestros', ascending=False)
conductor_detail['Tasa_Culpabilidad_%'] = (conductor_detail['Veces_Culpable'] / conductor_detail['Total_Siniestros'] * 100).round(1)

print(conductor_detail)


👤 TOP 20 CONDUCTORES CON MÁS SINIESTROS:
                                   Total_Siniestros  Veces_Culpable  \
Nombre Conductor                                                      
ARIAS RUEDA EDISON STEVE                          6               3   
DARIO RICARDO ROJAS RAMIREZ                       6               5   
CARDONA PEREZ YECIT                               6               3   
BLANDON MEJIA ALVARO DE JESUS                     6               2   
RIOS SALAZAR JOSE NELSON                          6               3   
SOSSA MEJIA DIEGO ALEJANDRO                       5               3   
BLANDON GARCIA ALVARO DE JESUS                    5               3   
TUBERQUIA OCAMPO SAMUEL ALEXANDER                 5               1   
FONTALVO PIZARRO JOSE LUIS                        5               5   
AREIZA DIAZ PEDRO ANDRES                          4               2   
ANGEL DE JESUS MARTINEZ VARGAS                    4               1   
DUQUE DURANGO JHONNATAN SNEIDER    

In [45]:
# Análisis de culpabilidad del conductor
culpabilidad_counts = df['Culpabilidad Conductor'].value_counts()

fig = make_subplots(
    rows=1, cols=2,
    subplot_titles=('Distribución de Culpabilidad', 'Culpabilidad por Causa'),
    specs=[[{'type': 'pie'}, {'type': 'bar'}]]
)

fig.add_trace(
    go.Pie(labels=culpabilidad_counts.index, values=culpabilidad_counts.values, hole=0.4),
    row=1, col=1
)

culpa_causa = pd.crosstab(df['Causa Siniestro'], df['Culpabilidad Conductor'], normalize='index') * 100

fig.add_trace(
    go.Bar(x=culpa_causa.index, y=culpa_causa['SI'], name='Culpable', marker_color='#e74c3c'),
    row=1, col=2
)

fig.update_layout(height=500, showlegend=True)
fig.show()

## ⚠️ 10. Análisis de Gravedad

In [46]:
# Distribución por gravedad
print("\n⚠️ DISTRIBUCIÓN POR NIVEL DE GRAVEDAD:")
print("="*80)

gravedad_counts = df['Gravedad'].value_counts()
gravedad_pct = (gravedad_counts / len(df)) * 100

gravedad_df = pd.DataFrame({
    'Cantidad': gravedad_counts,
    'Porcentaje': gravedad_pct
})
print(gravedad_df)

# Gráfico de gravedad
fig = go.Figure(data=[go.Pie(
    labels=gravedad_counts.index,
    values=gravedad_counts.values,
    hole=0.4,
    marker=dict(colors=['#2ecc71', '#3498db', '#f39c12', '#e67e22', '#e74c3c'])
)])

fig.update_layout(
    title='Distribución de Siniestros por Nivel de Gravedad',
    height=500
)
fig.show()


⚠️ DISTRIBUCIÓN POR NIVEL DE GRAVEDAD:
            Cantidad  Porcentaje
Gravedad                        
Leve             817       72.11
Sin tiempo       165       14.56
Moderado         138       12.18
Grave              7        0.62
Crítico            6        0.53


In [47]:
# Casos críticos (>100 horas)
casos_criticos = df[df['Gravedad'] == 'Crítico']

print(f"\n🚨 CASOS CRÍTICOS (>100 horas inoperativas): {len(casos_criticos)}")
print("="*80)

if len(casos_criticos) > 0:
    print("\nTop 10 casos más críticos:")
    top_criticos = casos_criticos.nlargest(10, 'Tiempo Inoperativo ( HORAS )')[
        ['Fecha Siniestro', 'DEPARTAMENTO', 'CIUDAD', 'Tipo de vehículo', 
         'Tipo accidente', 'Tiempo Inoperativo ( HORAS )', 'Causa Siniestro']
    ]
    print(top_criticos)


🚨 CASOS CRÍTICOS (>100 horas inoperativas): 6

Top 10 casos más críticos:
    Fecha Siniestro DEPARTAMENTO            CIUDAD Tipo de vehículo  \
840      2024-10-05        CESAR        VALLEDUPAR        Camioneta   
652      2022-09-20    ANTIOQUIA             BELLO           Camion   
65       2024-07-28    ATLANTICO  PALMAR DE VARELA         Volqueta   
701      2023-11-14        CESAR        VALLEDUPAR         Volqueta   
764      2025-09-15        CESAR            LA PAZ      Compactador   
853      2025-06-10        CESAR        VALLEDUPAR      Compactador   

                  Tipo accidente  Tiempo Inoperativo ( HORAS )  \
840  Daños a vehículo propio (No                          3600   
652                   Colisiones                          2592   
65                    Colisiones                           384   
701  Daños a vehículo propio (No                           362   
764   Pérdida total del vehículo                           256   
853                   Colisione

## 📊 11. Análisis Multivariado y Correlaciones

In [48]:
# Análisis de tablas cruzadas
print("\n📊 TABLAS CRUZADAS:")
print("="*80)

print("\n1️⃣ Turno vs Culpabilidad:")
tabla1 = pd.crosstab(df['Turno'], df['Culpabilidad Conductor'], margins=True)
print(tabla1)

print("\n2️⃣ Departamento vs Causa (Top causas):")
tabla2 = pd.crosstab(df['DEPARTAMENTO'], df['Causa Siniestro'], margins=True)
print(tabla2)

print("\n3️⃣ Tipo de Vehículo vs Uso de Grúa (Top vehículos):")
top_vehiculos = df['Tipo de vehículo'].value_counts().head(5).index
tabla3 = pd.crosstab(
    df[df['Tipo de vehículo'].isin(top_vehiculos)]['Tipo de vehículo'],
    df[df['Tipo de vehículo'].isin(top_vehiculos)]['Usó Grua']
)
print(tabla3)


📊 TABLAS CRUZADAS:

1️⃣ Turno vs Culpabilidad:
Culpabilidad Conductor   NO   SI   All
Turno                                 
Diurno                  399  289   688
Nocturno                225  220   445
All                     624  509  1133

2️⃣ Departamento vs Causa (Top causas):
Causa Siniestro  FACTORES EXTERNOS  FALLAS HUMANAS  FALLAS MTTO   All
DEPARTAMENTO                                                         
ANTIOQUIA                      241             277           18   536
ATLANTICO                       66              46            6   118
CESAR                          164              28           10   202
TOLIMA                         146             115           16   277
All                            617             466           50  1133

3️⃣ Tipo de Vehículo vs Uso de Grúa (Top vehículos):
Usó Grua           NO  SI
Tipo de vehículo         
Amplirolles        53   4
Buseta             35   1
Camion             39   5
Compactador       743  62
Volqueta        

In [49]:
# Heatmap de correlaciones entre variables categóricas
# Crear variables numéricas para correlación
df_corr = df.copy()
df_corr['Culpable_Num'] = (df_corr['Culpabilidad Conductor'] == 'SI').astype(int)
df_corr['Reportado_Num'] = (df_corr['Reportado Aseguradora'] == 'SI').astype(int)
df_corr['Grua_Num'] = (df_corr['Usó Grua'] == 'SI').astype(int)
df_corr['Juridica_Num'] = (df_corr['Asistencia Jurídica'] == 'SI').astype(int)
df_corr['Turno_Num'] = (df_corr['Turno'] == 'Nocturno').astype(int)

corr_cols = ['Tiempo Inoperativo ( HORAS )', 'Culpable_Num', 'Reportado_Num', 
             'Grua_Num', 'Juridica_Num', 'Turno_Num']

corr_matrix = df_corr[corr_cols].corr()

fig = go.Figure(data=go.Heatmap(
    z=corr_matrix.values,
    x=['Tiempo Inop', 'Culpable', 'Reportado', 'Grúa', 'Jurídica', 'Turno Noct'],
    y=['Tiempo Inop', 'Culpable', 'Reportado', 'Grúa', 'Jurídica', 'Turno Noct'],
    colorscale='RdBu',
    zmid=0,
    text=corr_matrix.values.round(2),
    texttemplate='%{text}',
    textfont={"size": 12}
))

fig.update_layout(
    title='Matriz de Correlaciones',
    height=600
)
fig.show()

## 📝 12. Conclusiones y Recomendaciones

### 🎯 Principales Hallazgos

**1. Distribución Geográfica:**
- Antioquia concentra el mayor número de siniestros (47.3%)
- Las ciudades de Ibagué, Bello y Valledupar lideran en incidentes

**2. Causas de Siniestros:**
- Los factores externos son la causa principal (54.5%)
- Las fallas humanas representan el 41.1% de los casos
- Las fallas de mantenimiento son minoritarias (4.4%)

**3. Patrones Temporales:**
- Mayor incidencia en turno diurno (60.7%)
- Los días laborables concentran más siniestros
- Se observa tendencia creciente en años recientes

**4. Vehículos:**
- Los compactadores están involucrados en el 71% de los casos
- Las volquetas tienen mayor tiempo inoperativo promedio

**5. Conductores:**
- El 44.9% de los casos tienen culpabilidad del conductor
- Existe un grupo de conductores reincidentes

### 💡 Recomendaciones Estratégicas

**1. Prevención:**
- Implementar capacitaciones focalizadas en ciudades críticas
- Reforzar protocolos de mantenimiento preventivo
- Mejorar inspecciones pre-operacionales

**2. Gestión de Riesgos:**
- Monitoreo especial a conductores reincidentes
- Evaluación periódica de competencias
- Programa de incentivos por seguridad

**3. Operaciones:**
- Optimizar rutas en zonas de alto riesgo
- Ajustar horarios en áreas críticas
- Mejorar comunicación con autoridades locales

**4. Tecnología:**
- Implementar sistemas de telemetría
- Monitoreo en tiempo real de flotas
- Alertas tempranas de condiciones de riesgo

**5. Seguimiento:**
- Dashboard de indicadores en tiempo real
- Reportes automáticos de tendencias
- Auditorías periódicas de cumplimiento


**Fecha de análisis:** Marzo 2025

**Responsable:** Duber Alexander Zapata Higuita
                 , Mauricio Giraldo
                 , Sebastian Pineda Alvarez   
